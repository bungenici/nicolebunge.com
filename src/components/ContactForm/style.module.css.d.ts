/* eslint-disable @typescript-eslint/camelcase */

export const form: string;
export const form__field: string;
export const form__input: string;
export const form__label: string;
