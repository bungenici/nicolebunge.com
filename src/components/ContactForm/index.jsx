import React from 'react';
import { Formik, Form, Field } from 'formik';
import GridColumn from '../GridColumn';
import Label from '../Label';
import GridRow from '../GridRow';
import styles from './style.module.css';

const ContactForm = () => (
  <Formik>
    <Form
      action="/kontakt/danke"
      data-netlify-honeypot="bot-field"
      data-netlify="true"
      className={styles.form}
      method="post"
      name="contact"
    >
      <Field name="form-name" type="hidden" value="contact" />

      <GridRow>
        <GridColumn columnSpan={6}>
          <div className={styles.form__field}>
            <Field
              autoComplete="name"
              className={styles.form__input}
              name="name"
              placeholder="Name"
              required
              type="text"
            />
            <Label className={styles.form__label} htmlFor="name">
              Name
            </Label>
          </div>
        </GridColumn>

        <GridColumn columnSpan={6} columnStart={7}>
          <div className={styles.form__field}>
            <Field
              autoComplete="email"
              className={styles.form__input}
              name="email"
              placeholder="E-Mail"
              required
              type="email"
            />
            <Label className={styles.form__label} htmlFor="email">
              E-Mail
            </Label>
          </div>
        </GridColumn>

        <GridColumn columnSpan={12}>
          <div className={styles.form__field}>
            <Field
              className={styles.form__input}
              component="textarea"
              name="message"
              placeholder="Nachricht"
              required
            />
            <Label className={styles.form__label} htmlFor="message">
              Nachricht
            </Label>
          </div>
        </GridColumn>
      </GridRow>

      <div className={styles.form__field}>
        <button type="submit">Senden</button>
      </div>
    </Form>
  </Formik>
);

export default ContactForm;
