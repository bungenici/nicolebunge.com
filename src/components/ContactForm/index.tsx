import { Field, Form, Formik } from 'formik';
import React from 'react';
import GridColumn from '../GridColumn';
import Label from '../Label';
import Row from '../Row';
import styles from './style.module.css';

const ContactForm: React.FC = () => (
  <Formik
    initialValues={{ name: '', email: '', message: '' }}
    onSubmit={(_values, { setSubmitting }): void => {
      setSubmitting(false);
    }}
  >
    {({ isSubmitting }): JSX.Element => (
      <Form
        action="/kontakt/danke"
        data-netlify-honeypot="bot-field"
        data-netlify="true"
        className={styles.form}
        method="post"
        name="contact"
      >
        <Field name="form-name" type="hidden" value="contact" />

        <Row>
          <GridColumn columnSpan={6}>
            <div className={styles.form__field}>
              <Field
                autoComplete="name"
                className={styles.form__input}
                name="name"
                placeholder="Name"
                required
                type="text"
              />
              <Label className={styles.form__label} htmlFor="name">
                Name
              </Label>
            </div>
          </GridColumn>

          <GridColumn columnSpan={6} columnStart={7}>
            <div className={styles.form__field}>
              <Field
                autoComplete="email"
                className={styles.form__input}
                name="email"
                placeholder="E-Mail"
                required
                type="email"
              />
              <Label className={styles.form__label} htmlFor="email">
                E-Mail
              </Label>
            </div>
          </GridColumn>

          <GridColumn columnSpan={12}>
            <div className={styles.form__field}>
              <Field
                className={styles.form__input}
                component="textarea"
                name="message"
                placeholder="Nachricht"
                required
              />
              <Label className={styles.form__label} htmlFor="message">
                Nachricht
              </Label>
            </div>
          </GridColumn>
        </Row>

        <div className={styles.form__field}>
          <button type="submit" disabled={isSubmitting}>
            Senden
          </button>
        </div>
      </Form>
    )}
  </Formik>
);

export default ContactForm;
