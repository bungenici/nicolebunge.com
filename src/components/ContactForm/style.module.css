.form__field {
  margin-bottom: 1.5em;
  margin-top: 0.5em;
}

.form__input {
  border: 1px solid #000;
  padding: 0.5em 1em;
  width: 100%;
}
