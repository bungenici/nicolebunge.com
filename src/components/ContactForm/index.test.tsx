/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import ContactForm from '.';

describe('ContactForm component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ContactForm />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
