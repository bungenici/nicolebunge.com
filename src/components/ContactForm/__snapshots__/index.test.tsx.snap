// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component renders correctly 1`] = `
<form
  action="/kontakt/danke"
  className="form"
  data-netlify="true"
  data-netlify-honeypot="bot-field"
  method="post"
  name="contact"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    name="form-name"
    onBlur={[Function]}
    onChange={[Function]}
    type="hidden"
    value="contact"
  />
  <div
    className="grid__row"
  >
    <div
      className="grid__column grid__columnColumnSpan6 grid__columnColumnStart1"
    >
      <div
        className="form__field"
      >
        <input
          autoComplete="name"
          className="form__input"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
        <label
          className="form__label label"
          htmlFor="name"
        >
          Name
        </label>
      </div>
    </div>
    <div
      className="grid__column grid__columnColumnSpan6 grid__columnColumnStart7"
    >
      <div
        className="form__field"
      >
        <input
          autoComplete="email"
          className="form__input"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="E-Mail"
          required={true}
          type="email"
          value=""
        />
        <label
          className="form__label label"
          htmlFor="email"
        >
          E-Mail
        </label>
      </div>
    </div>
    <div
      className="grid__column grid__columnColumnSpan12 grid__columnColumnStart1"
    >
      <div
        className="form__field"
      >
        <textarea
          className="form__input"
          name="message"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Nachricht"
          required={true}
          value=""
        />
        <label
          className="form__label label"
          htmlFor="message"
        >
          Nachricht
        </label>
      </div>
    </div>
  </div>
  <div
    className="form__field"
  >
    <button
      disabled={false}
      type="submit"
    >
      Senden
    </button>
  </div>
</form>
`;
